### PR TITLE
API-2922 Update model and add some dummy endpoints

### PR DESCRIPTION
**Pull request by Tobias1087**

### CODE REVIEW

File: app/uk/gov/hmrc/apinotificationqueue/controller/Queue.scala [hunk 1]
```diff
@@ -16,18 +16,31 @@
...
-import uk.gov.hmrc.play.bootstrap.controller.BaseController
+import play.api.http.HttpEntity
 import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.controller.BaseController
 
 import scala.concurrent.Future
 
 @Singleton()
-class MicroserviceHelloWorld extends BaseController {
+class Queue extends BaseController {
```
Comment (gokyo) [inline app/uk/gov/hmrc/apinotificationqueue/controller/Queue.scala]: `QueueController` perhaps?!

File: app/uk/gov/hmrc/apinotificationqueue/repository/Message.scala [hunk 2]
```diff
@@ -18,4 +18,5 @@ package uk.gov.hmrc.apinotificationqueue.repository
 
 import java.util.UUID
 
-case class Message(messageId: UUID, clientId: String, subscriptionFieldsId: String)
+
+case class Message(messageId: UUID, headers: Map[String, String], payload: String, dateReceived: String)
```
Comment (gokyo) [inline app/uk/gov/hmrc/apinotificationqueue/repository/Message.scala]: Is it not better to use a timestamp/date type for the `dateReceived` ?

File: app/uk/gov/hmrc/apinotificationqueue/repository/MessageRepository.scala [hunk 3]
```diff
@@ -21,7 +21,11 @@ import java.util.UUID
 import scala.concurrent.Future
 
 trait MessageRepository {
```
Comment (gokyo) [inline app/uk/gov/hmrc/apinotificationqueue/repository/MessageRepository.scala]: `clientId` is String, not a UUID

File: test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala [hunk 4]
```diff
@@ -18,22 +18,33 @@ package uk.gov.hmrc.apinotificationqueue.controller
...
     "return 200" in {
-      val controller = new MicroserviceHelloWorld()
-      val result = controller.hello()(fakeRequest)
+      val controller = new Queue()
+      val result = controller.getAll()(FakeRequest("GET", "/messages"))
       status(result) shouldBe Status.OK
     }
   }
 
+  "GET /message/:id" should {
```
Comment (gokyo) [inline test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala]: Shall we add also a 404 case?

Comment (Tobias1087) [reply test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala]: It's currently hard coded to return so when we actually do the logic yes.

File: app/uk/gov/hmrc/apinotificationqueue/repository/MessageRepository.scala [hunk 5]
```diff
@@ -21,7 +21,11 @@ import java.util.UUID
...
-  def save(message: Message): Future[Message]
-  def fetch(messageId: UUID): Future[Option[Message]]
-  def delete(messageId: UUID): Future[Boolean]
+  def save(clientId: UUID, message: Message): Future[Message]
+
```
Comment (gokyo) [inline app/uk/gov/hmrc/apinotificationqueue/repository/MessageRepository.scala]: In all operations, the parameter `clientId` should be a `String`, not a `UUID`.

File: test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala [hunk 6]
```diff
@@ -18,22 +18,33 @@ package uk.gov.hmrc.apinotificationqueue.controller
...
-import uk.gov.hmrc.play.test.WithFakeApplication
 import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 
-class MicroserviceHelloWorldControllerSpec extends UnitSpec with WithFakeApplication{
+class QueueControllerSpec extends UnitSpec with WithFakeApplication {
 
-  val fakeRequest = FakeRequest("GET", "/")
 
-  "GET /" should {
+  "GET /messages" should {
```
Comment (gokyo) [inline test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala]: In my opinion this should return 200 also in the case of an empty list

File: test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala [hunk 7]
```diff
@@ -18,22 +18,33 @@ package uk.gov.hmrc.apinotificationqueue.controller
...
 import play.api.http.Status
 import play.api.test.FakeRequest
-import play.api.http.Status
-import play.api.test.FakeRequest
-import uk.gov.hmrc.play.test.UnitSpec
-import uk.gov.hmrc.play.test.WithFakeApplication
 import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 
-class MicroserviceHelloWorldControllerSpec extends UnitSpec with WithFakeApplication{
+class QueueControllerSpec extends UnitSpec with WithFakeApplication {
```
Comment (gokyo) [inline test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala]: Do we need also a "get message by clientId" case?

Comment (Tobias1087) [reply test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala]: I don't think so as we don't have that functionality and won't need it with the change to have a GET All then a DELETE by ID

File: test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala (hunk 6 above)
Comment (Tobias1087) [reply test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala]: We don't currently have this functionality when we do we should you are correct.

**Pull request merged**